### PR TITLE
Simplify backend code

### LINKED ISSUE #2453: Use struct inheritance to simplify backend code
Like in #2438.

### LINKED ISSUE #3595: Drop unused fields from detail structures or use them as initial list data for details view
At the moment a lot of detail structures like `DeploymentDetails` have a list of connected resources inside them but we do not use them. It is a waste. We should use them or remove them from the structures. I'd vote for removing them and keeping it like it is at the moment.

### PR DESCRIPTION
**Pull request by maciaszczykm**
- Drops unused fields from detail structs (closes #3595).
- Uses struct inheritance to simplify detail structs (closes #2453).
- Aligns classes in `backendapi.d.ts` with the backend.
- Switch to apps/v1.

### CODE REVIEW

File: src/app/backend/resource/service/list_test.go [hunk 1]
```diff
@@ -15,6 +15,8 @@
 package service
 
 import (
+	"github.com/kubernetes/dashboard/src/app/backend/resource/endpoint"
```
Comment (shu-mutou) [inline src/app/backend/resource/service/list_test.go]: These two imports should be after L25 ("github.com/kubernetes/dashboard/src/app/backend/resource/dataselect")

**Comment by maciaszczykm** [comment]
@jeefy @floreks It would be awesome if one of you would check it 🙂

**Comment by jeefy** [comment]
This looks awesome and simplifies a bit. 👍 

Once you made that package.json change you addressed the only thing I had a question about. (I was mid-review and saw the email lol)

Passes automated and manual/local testing. Happy we're finally updating to AppsV1 as well 🎉 

@maciaszczykm I'll lgtm unless you have any other last minute changes.

**Comment by maciaszczykm** [comment]
@jeefy No other changes :)

**Pull request merged**